ray-operator/controllers/ray/utils/util_test.go

```diff
@@ -840,3 +948,226 @@ func TestIsGCSFaultToleranceEnabled(t *testing.T) {
 		})
 	}
 }
+
+func createRayClusterTemplate(
+	head struct {
+		cpu    string
+		memory string
+	},
+	workers []struct {
+		replicas    *int32
+		minReplicas *int32
+		suspend     *bool
+		cpu         string
+		memory      string
+		numOfHosts  int32
+	},
+) *rayv1.RayCluster {
+	cluster := &rayv1.RayCluster{
+		Spec: rayv1.RayClusterSpec{
+			HeadGroupSpec: rayv1.HeadGroupSpec{
+				Template: corev1.PodTemplateSpec{
+					Spec: corev1.PodSpec{
+						Containers: []corev1.Container{
+							{
+								Resources: corev1.ResourceRequirements{
+									Requests: corev1.ResourceList{
+										corev1.ResourceCPU:    resource.MustParse(head.cpu),
+										corev1.ResourceMemory: resource.MustParse(head.memory),
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+
+	for _, w := range workers {
+		cluster.Spec.WorkerGroupSpecs = append(cluster.Spec.WorkerGroupSpecs, rayv1.WorkerGroupSpec{
+			NumOfHosts:  w.numOfHosts,
+			Replicas:    w.replicas,
+			MinReplicas: w.minReplicas,
+			Suspend:     w.suspend,
+			Template: corev1.PodTemplateSpec{
+				Spec: corev1.PodSpec{
+					Containers: []corev1.Container{
+						{
+							Resources: corev1.ResourceRequirements{
+								Requests: corev1.ResourceList{
+									corev1.ResourceCPU:    resource.MustParse(w.cpu),
+									corev1.ResourceMemory: resource.MustParse(w.memory),
+								},
+							},
+						},
+					},
+				},
+			},
+		})
+	}
+
+	return cluster
+}
+
+func TestCalculateDesiredResources(t *testing.T) {
```

Comment:
How about using something like “100Mi” instead of “128Mi” so I can verify the answer more easily?